src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,671 @@
+				$style_attribute_value     = $p->get_attribute( 'style' );
+				$style_attribute_value     = ( $style_attribute_value && ! is_bool( $style_attribute_value ) ) ? $style_attribute_value : '';
+
+				/*
+				 * Checks first if the style property is not falsy and the style
+				 * attribute value is not empty because if it is, it doesn't need to
+				 * update the attribute value.
+				 */
+				if ( $style_property_value || ( ! $style_property_value && $style_attribute_value ) ) {
+					$style_attribute_value = $this->set_style_property( $style_attribute_value, $style_property, $style_property_value );
+					/*
+					 * If the style attribute value is not empty, it sets it. Otherwise,
+					 * it removes it.
+					 */
+					if ( ! empty( $style_attribute_value ) ) {
+						$p->set_attribute( 'style', $style_attribute_value );
+					} else {
+						$p->remove_attribute( 'style' );
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Sets an individual style property in the `style` attribute of an HTML
+	 * element, updating or removing the property when necessary.
+	 *
+	 * If a property is modified, it is added at the end of the list to make sure
+	 * that it overrides the previous ones.
+	 *
+	 * @since 6.5.0
+	 *
+	 * Example:
+	 *
+	 *     set_style_property( 'color:green;', 'color', 'red' )      => 'color:red;'
+	 *     set_style_property( 'background:green;', 'color', 'red' ) => 'background:green;color:red;'
+	 *     set_style_property( 'color:green;', 'color', null )       => ''
+	 *
+	 * @param string            $style_attribute_value The current style attribute value.
+	 * @param string            $style_property_name   The style property name to set.
+	 * @param string|false|null $style_property_value  The value to set for the style property. With false, null or an
+	 *                                                 empty string, it removes the style property.
+	 * @return string The new style attribute value after the specified property has been added, updated or removed.
+	 */
+	private function set_style_property( string $style_attribute_value, string $style_property_name, $style_property_value ): string {
+		$style_assignments    = explode( ';', $style_attribute_value );
+		$result               = array();
+		$style_property_value = ! empty( $style_property_value ) ? rtrim( trim( $style_property_value ), ';' ) : null;
+		$new_style_property   = $style_property_value ? $style_property_name . ':' . $style_property_value . ';' : '';
+
+		// Generate an array with all the properties but the modified one.
+		foreach ( $style_assignments as $style_assignment ) {
+			if ( empty( trim( $style_assignment ) ) ) {
+				continue;
+			}
+			list( $name, $value ) = explode( ':', $style_assignment );
+			if ( trim( $name ) !== $style_property_name ) {
+				$result[] = trim( $name ) . ':' . trim( $value ) . ';';
+			}
+		}
+
+		// Add the new/modified property at the end of the list.
+		array_push( $result, $new_style_property );
```

Comment:
```suggestion
		$result[] = $new_style_property;
```

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5953/commits/19e454fcea683c1cd1efa8909c0ae3e0330195aa.